docs/advanced-usage/connecting-cloud-storage/amazon-s3.md

```diff
@@ -0,0 +1,111 @@
+Integrations can be established to Amazon S3.
+
+To get started, ensure you have admin access within Kolena.
+Navigate to your [Organization Settings's Integrations Tab](https://app.kolena.io/redirect/organization?tab=integrations) and click "Add Integration", then "AWS".
+
+### 1. Allow CORS Access to Bucket
+
+CORS permissions are required for the Kolena domain to render content from your bucket.
+
+Navigate to your S3 bucket inside your AWS console.
+Click on the "Permissions" tab and navigate to the "Cross-origin resource sharing (CORS)" section.
+Click "Edit" and add the following JSON snippet:
+
+```json
+[
+  {
+    "AllowedHeaders": ["*"],
+    "AllowedMethods": ["GET"],
+    "AllowedOrigins": ["https://app.kolena.io"],
+    "ExposeHeaders": []
+  }
+]
+```
+
+### 2. Create an Access Policy in AWS
+
+First ensure you have IAM write permissions within your AWS account.
+In your AWS console, navigate to the IAM policies page.
+Click the "Create Policy" button and select the "JSON" tab.
+
+Paste the following JSON policy (please replace the bucket name `share-with-kolena` with the appropriate bucket)
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "S3ListBucket",
+      "Effect": "Allow",
+      "Action": ["s3:GetObject", "s3:ListBucket"],
+      "Resource": [
+        "arn:aws:s3:::share-with-kolena",
+        "arn:aws:s3:::share-with-kolena/*"
+      ]
+    }
+  ]
+}
+```
+
+Click through the "Next" buttons, adding the desired name, description, and tags.
+
+### 3. Create a Role For Kolena to Assume
+
+Return to the Kolena platform [Integrations Tab](https://app.kolena.io/redirect/organization?tab=integrations).
+
+On the "Create Amazon S3 Integration" page, click "Generate a Principal ARN".
+This will create a Principal which will be referenced in a trust policy.
+If desired, an External Id may also be provided to include within a trust policy.
+
+Once these steps are complete, copy the JSON which appears on the page.
+This will be of the form
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "KolenaAssumeRole",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "<< Principal ARN >>"
+      },
+      "Action": "sts:AssumeRole",
+      "Condition": {
+        "StringEquals": {
+          "sts:ExternalId": ["<< External Id >>"]
+        }
+      }
```

Comment:
What happens here when users don't specify an external ID?

Reply:
This condition block doesn't appear. The integrations page handles that logic for users when generating the JSON